src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a >= this.b + this.c
+      || this.b >= this.a + this.c
+      || this.c >= this.b + this.a
+      || this.a <= 0
+      || this.b <= 0
+      || this.c <= 0) {
+      throw new Error('The length of triangle is not correct');
+    }
+  }
 
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return +(Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c))
+      .toFixed(2));
+  }
 }
 
 export class Circle {
+  shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('the radius is not correct');
+    }
```

Comment:
will be better to create separate function that will accept input arguments and will check if each of them is bigger than 0
and call this function for each figure